### PR TITLE
fix(assets): Fallback format not being taken into account properly

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

What the title says.

This PR also adds an error message when converting from SVGs to raster formats because it's easier to do by accident on the Picture component

## Testing

Added a test

## Docs

N/A apart from the new error

### CODE REVIEW

File: packages/astro/src/core/errors/errors-data.ts [hunk 1]
```diff
@@ -554,6 +554,21 @@ export const UnsupportedImageFormat = {
 		)} are supported by our image services.`,
 	hint: "Using an `img` tag directly instead of the `Image` component might be what you're looking for.",
 } satisfies ErrorData;
+
+/**
+ * @docs
+ * @see
+ * - [Images](https://docs.astro.build/en/guides/images/)
+ * @description
+ * Astro does not currently supporting converting between vector (such as SVGs) and raster (such as PNGs and JPEGs) images.x
```
Comment (florian-lefebvre) [inline packages/astro/src/core/errors/errors-data.ts:563]:
```suggestion
 * Astro does not currently supporting converting between vector (such as SVGs) and raster (such as PNGs and JPEGs) images.
```

File: packages/astro/components/Picture.astro [hunk 2]
```diff
@@ -24,16 +24,14 @@ const optimizedImages: GetImageResult[] = await Promise.all(
...
 );
 
-const fallbackFormat =
-	props.fallbackFormat ?? isESMImportedImage(props.src)
-		? ['svg', 'gif'].includes(props.src.format)
-			? props.src.format
-			: 'png'
-		: 'png';
+let resultFallbackFormat = fallbackFormat ?? 'png';
+if (isESMImportedImage(props.src) && ['gif', 'svg'].includes(props.src.format)) {
```
Comment (ematipico) [inline packages/astro/components/Picture.astro:28]: nit: `['gif', 'svg']` could be moved into a top-level variable

**Review by ematipico — APPROVED** [review]

File: .changeset/dull-bikes-decide.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fix Picture component not taking into account the fallback format specified
```
Comment (sarah11918) [inline .changeset/dull-bikes-decide.md:5]:
```suggestion
Fixes Picture component not taking into account the fallback format specified
```

**Review by sarah11918 — APPROVED** [review]
Just standardizing our verb tense for the changeset, otherwise, docs approves!

I realized we don't have this guidance in our actual image page in docs, so I'm making myself a task to add that info in.

File: packages/astro/components/Picture.astro (hunk 2 above)
Comment (itsmatteomanf) [reply packages/astro/components/Picture.astro:28]: I'm gonna agree here, it took me a second to understand what they did while investigating the issue on the previous code, and it's the same here. A tiny bit better, but a separate variable with a descriptive name would go along way.

Comment (Princesseuh) [reply packages/astro/components/Picture.astro:28]: Sure, we can do that

**Pull request merged**